src/main/scala/gov/wicourts/json/formlet/Forms.scala

```diff
@@ -2,55 +2,48 @@ package gov.wicourts.json.formlet
 
 import argonaut._
 import argonaut.Argonaut._
-
+import cats.Applicative
+import cats.Endo
+import cats.Functor
+import cats.Id
+import cats.data.Kleisli
+import cats.data.NonEmptyList
+import cats.data.OptionT
+import cats.data.State
+import cats.data.Validated
+import cats.syntax.all._
 import org.slf4j.LoggerFactory
-
-import scalaz._
-import scalaz.OptionT.optionT
-import scalaz.std.list._
-import scalaz.std.option._
-import scalaz.std.string._
-import scalaz.std.tuple._
-import scalaz.syntax.applicative._
-import scalaz.syntax.bind._
-import scalaz.syntax.either._
-import scalaz.syntax.equal._
-import scalaz.syntax.std.option._
-import scalaz.syntax.traverse._
-
-import scalaz.Id.Id
-
 import scala.language.higherKinds
 
 import Predef.identity
 
 object Forms {
   private val logger = LoggerFactory.getLogger(loggerName)
 
-  private def loggerName: String = getClass.getName.substring(0, getClass.getName.length-1)
+  private def loggerName: String = getClass.getName.substring(0, getClass.getName.length - 1)
 
   private def primitive[M[_]: Applicative, A](
     descr: String,
     toJson: A => Json,
     matches: Json => Boolean,
-    fromJson: Json => NonEmptyList[String] \/ A,
+    fromJson: Json => Either[NonEmptyList[String], A],
     name: String,
-    value: Option[A]
+    value: Option[A],
   ): FieldFormlet[M, Option[A]] =
     Formlet { c =>
-      val result = optionT[NonEmptyList[String] \/ ?](
-        c
-          .flatMap(_.downField(name))
+      val result = OptionT( //optionT[Either[NonEmptyList[String], *]](
+        c.flatMap(_.downField(name))
           .map(_.focus)
           .filterNot(j => j.isNull || j.string.exists(_ === ""))
-          .right[NonEmptyList[String]]
+          .asRight[NonEmptyList[String]],
       ).flatMapF(j =>
-        j.right[NonEmptyList[String]].ensure(
-          NonEmptyList(s"Field $name must be a(n) $descr")
-        )(
-          matches
-        ) >>= (fromJson)
-      ).run.validation
+          j.asRight[NonEmptyList[String]]
+            .ensure(NonEmptyList.of(s"Field $name must be a(n) $descr"))(matches)
+            .flatMap(fromJson)
+            .map(_.some),
```

Comment:
Why is this necessary?

Reply:
If you figure it out, I'd love to know. They're defined almost identically, but this also came up in the ccap3 conversion, with a flatMapF that compiled in scalaz despite there being no Options around to use. Seems like it ought to have been broken from the start, since none of right, ensure or fromJson have any way of providing an Option to satisfy flatMapF

Reply:
Can we do away with `OptionT` and the `.map(_.some)`?

Reply:
done

Reply:
Looks good.